algebird-core/src/main/scala/com/twitter/algebird/MinHasher.scala

```diff
@@ -130,6 +140,12 @@ abstract class MinHasher[H](val numHashes: Int, val numBands: Int)(implicit n: N
 
   /** Decode two signatures into hash values, combine them somehow, and produce a new array */
   protected def buildArray(left: Array[Byte], right: Array[Byte])(fn: (H, H) => H): Array[Byte]
+
+  /**
+   * Decode multiple signatures into hash values and combine them using given fn.
+   * This version is used by the new similarityMulti.
+   */
+  protected def buildArrayMulti(buffers: Seq[Array[Byte]])(fn: Seq[H] => H): Array[Byte]
```

Comment:
if you implement this as a private method on the companion that matches the case class we have, I think we can make this binary compatible.

```scala
private def buildArrayMulti[H](h: MinHasher[H], hasherbuffers: Seq[Array[Byte]])(fn: Seq[H] => H): Array[Byte] =
  h match {
```

Reply:
Thanks for reviewing.  Actually, buildArrayMulti is an abstract method called by  calls other buildArray virtual method.  If I move it to companion object, I'll have to move all of them to companion object.  Does it make sense to extend MinHasher32 and implement a new MinHasher32MultiSig with the new methods (same for 16bit version)?
```
class MinHasher32MultiSig(numHashes: Int, numBands: Int)(implicit n: Numeric[Int])
  extends MinHasher32(numHashes, numBands) {
  protected def buildArrayMulti(buffers: Seq[Array[Byte]])(fn: Vector[Int] => Int): Array[Byte] = {
    val intBuffers = buffers.map(b => ByteBuffer.wrap(b).asIntBuffer)
    buildArray(fn(intBuffers.map(_.get).toVector))
  }

  def similarityMulti(sigs: MinHashSignature*): Double = {
    buildArrayMulti(sigs.map(_.bytes))(vals => if (vals.forall(_ =? vals.head)) n.one else n.zero)
      .map(_.toDouble)
      .sum / numHashes
  }
}```